addons/web_editor/controllers/main.py

```diff
@@ -199,11 +199,15 @@ def video_url_data(self, video_url, autoplay=False, loop=False,
         )
 
     @http.route('/web_editor/attachment/add_data', type='json', auth='user', methods=['POST'], website=True)
-    def add_data(self, name, data, is_image, quality=0, width=0, height=0, res_id=False, res_model='ir.ui.view', **kwargs):
+    def add_data(self, name, data, is_image, quality=0, width=0, height=0, res_id=False, res_model='ir.ui.view', is_website=False, **kwargs):
         data = b64decode(data)
+        context = None
         if is_image:
             format_error_msg = _("Uploaded image's format is not supported. Try with: %s", ', '.join(SUPPORTED_IMAGE_EXTENSIONS))
             try:
+                if (is_website and request.env['ir.config_parameter'].sudo().get_param('website.no_quality_optimization', '')) or request.env['ir.config_parameter'].sudo().get_param('global.no_quality_optimization', ''):
+                    quality = -1 # disable image quality optimization
+                    context = {'image_no_postprocess': True}
```

Comment:
This logic should be moved inside `_attachment_create`. Instead of passing a random `context` dict, pass whatever specialized parameters you need.

Also why not make a separate route for the website variant, instead of burying this under more query params. Wouldn't it make sense to have a route like `/web_editor/media/upload` that specifically means you're uploading a website media/image?

The actual method that implements it could be the same:

```py
@http.route('/web_editor/attachment/add_data', ...)
def add_data(self, name, ...):
    return _attachment_create(...)

@http.route('/web_editor/media/upload', ...)
def add_data(self, name, ...):
    return _attachment_create(..., website_media=True)

```